include/podio/UserDataCollection.h

```diff
@@ -0,0 +1,169 @@
+#ifndef USERDATACOLLECTION_H
+#define USERDATACOLLECTION_H
+
+#include "podio/CollectionBase.h"
+
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+#include <typeindex>
+
+
+namespace podio {
+
+
+  ///helper class providing names for basic types used in UserDataCollection<T>
+  class  UserDataTypes{
+  private:
+    UserDataTypes() = default ;
+    UserDataTypes(const UserDataTypes& ) = delete;
+    UserDataTypes& operator=(const UserDataTypes& ) = delete;
+    ~UserDataTypes() = default;
+
+    const std::map< const std::type_index, const std::string >_typeMap =
+    {
+      { std::type_index( typeid(int) ),   "podio::int"   },
+      { std::type_index( typeid(long) ),  "podio::long"  },
+      { std::type_index( typeid(float) ), "podio::float" },
+      { std::type_index( typeid(double) ),"podio::double"}
+    } ;
+
+  public:
+
+    std::string name( std::type_index idx ){
+      auto it = _typeMap.find( idx ) ;
+      return it != _typeMap.end() ? it->second  :  "UNKNOWN"  ;
+    }
+
+    // typename w/ 'podio::' replaced by 'podio_'
+    std::string sio_name( std::type_index idx ){
+      return std::string("podio_") + name( idx ).substr( 7 , 1024)  ;
+    }
+
+    static UserDataTypes& instance(){
+      static UserDataTypes me ;
+      return me ;
+    }
+  } ;
+
+
+  /// Templated base class for storing std::vectors of basic types as user data in PODIO
+  template <typename BasicType>
```

Comment:
This template should be constrained to only accept those types that we actually support via the mapping above. Currently it is possible to do something like
```cpp
auto& coll = store.create<podio::UserDataCollection<unsigned>>("unsignedColl");
writer.registerForWrite("unsignedColl");
```
which compiles fine but crashes at some point at run-time. It should be possible to catch this at compile time already either via some `static_assert` or `enable_if` trickery.

Reply:
Fully agreed, if we decide to only provide a fixed set of types - if not, then of course it cannot be done and we leave failure to runtime...